network/80-qubes.conf

```diff
@@ -1 +1,12 @@
-net.ipv6.conf.all.drop_unsolicited_na=1
+net.ipv4.conf.*.accept_source_route = 0
+net.ipv4.conf.*.accept_redirects = 0
+net.ipv4.conf.*.secure_redirects = 0
+net.ipv4.conf.*.send_redirects = 0
+net.ipv4.conf.*.drop_unicast_in_l2_multicast = 1
+net.ipv6.conf.*.accept_source_route = 0
+net.ipv6.conf.*.accept_redirects = 0
+net.ipv6.conf.*.accept_ra = 0
+net.ipv6.conf.*.accept_dad = 0
+net.ipv6.conf.*.autoconf = 0
+net.ipv6.conf.*.drop_unicast_in_l2_multicast = 1
+net.ipv6.conf.*.drop_unsolicited_na = 1
```

Comment:
This is rather massive number of changed settings, too big risk to get it into R4.1 at this point.

Reply:
Most of (all but `net.ipv[46].conf.*.drop_unicast_in_l2_multicast`) come from @fepitre’s [standard hardening configuration for CentOS][1], which in turn comes from the CIS CentOS 7 benchmark.  So I consider these to be very low-risk.  `net.ipv[46].conf.*.drop_unicast_in_l2_multicast` causes Linux to drop bogus packets that do not conform to standards, but it is the least important and I can get rid of it.

[1]: https://github.com/fepitre/centos-builder/blob/master/centos-hardened.ks#L192-L212

Reply:
It looks like this file is installed only on Debian system. I don't know why but for me it looks lie a duplicate with respect to `setup-ip` and `vif-route-qubes`.

Reply:
> However, I am not aware of any way vif-route-qubes could do that.

`vif-route-qubes` is called by `xendriverdomain` service, as part of bringing up `vif` interface. And `systemd-sysctl` is guaranteed to start earlier, since it starts before `sysinit.target`.

Reply:
To elaborate: I meant *for the specific network interface*.  It is trivial to set the `all` and `default` entries, but not the interface-specific ones.  If we had `Requires=systemd-sysctl.service` on `xendriverdomain.service`, that would likely be enough.  Should I do this with sysctl drop-ins, the script, or both?

Reply:
>  If we had Requires=systemd-sysctl.service

That's not necessary. `systemd-sysctl.service` is started always (it is static unit). If you mean ordering, then that's common misconception about dependencies in systemd - `Requires` is about _if_ the unit will be started, not _when_.

Reply:
In any case, please decide on one solution - either sysctl, or scripts configuring specific interfaces, but not both. I'd prefer sysctl, because it's possible to override it for the user, if specific use case would require rolling back some of those settings.